velox/functions/prestosql/DateTimeFunctions.h

```diff
@@ -1142,6 +1142,28 @@ struct ParseDateTimeFunction {
   }
 };
 
+template <typename T>
+struct CurrentTimeStampFunction : public TimestampWithTimezoneSupport<T> {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  int16_t timezoneId;
+
+  FOLLY_ALWAYS_INLINE void initialize(const core::QueryConfig& config) {
+    auto sessionTzName = config.sessionTimezone();
+    std::optional<int64_t> sessionTzID;
+    // Set the timezone to the session timezone. If there's
+    // no session timezone, fallback to 0 (GMT).
+    if (!sessionTzName.empty()) {
+      sessionTzID = util::getTimeZoneID(sessionTzName);
+    }
+    timezoneId = sessionTzID.value_or(0);
+  }
+
+  FOLLY_ALWAYS_INLINE void call(out_type<TimestampWithTimezone>& result) {
+    result = std::make_tuple(Timestamp::now().toMillis(), timezoneId);
```

Comment:
@yiweiHeOSS Is this the expected semantics? The description says that 
> Returns the current timestamp as of the start of the query.
Which means it should return the exact same timestamp on every call() which should be the time when the query started. however, this would instead return a new value every time

Reply:
Looked through the presto java implementation, looks like it used the "session start time" which aligns with the description of the functions. See https://github.com/prestodb/presto/blob/8499cd35657930a71d891eb0a5ef0d5917735ba1/presto-main/src/main/java/com/facebook/presto/operator/scalar/DateTimeFunctions.java#L159C60-L159C60

Can you please make the appropriate change?

Reply:
@bikramSingh91 Coverage maps should be updated in a separate PR.

Reply:
@bikramSingh91 : Presto evaluates current_timestamp at the co-ordinator and it never issues a Velox call for it. We added Velox implementation for completeness. There isn't a notion of "session start time" in Velox, so this current implementation was decided.

Reply:
@aditi-pandit Not sure it makes sense to add these functions. If there is no use case for these, then let's not add them.

Reply:
Thanks @mbasmanova. 

Agree this function is not worthwhile in Velox. There were previous discussions where the thinking was otherwise. Closing the PR.